pantheon-bundle/src/main/java/com/redhat/pantheon/extension/HydraIntegration.java

```diff
@@ -104,11 +113,26 @@ public void processEvent(Event event) throws Exception {
         MessageProducer producer = session.createProducer(session.createTopic(HYDRA_TOPIC));
         String moduleUUID = module.getValueMap().get(UUID_FIELD, String.class);
         String eventValue = ModuleVersionPublishedEvent.class.equals(event.getClass()) ? EVENT_PUBLISH_VALUE : EVENT_UNPUBLISH_VALUE;
-        String msg = "{\""
-                + ID_KEY + "\":" + "\"" + this.getPantheonHost() + PANTHEON_MODULE_API_PATH + moduleUUID +"\","
-                + "\"" + EVENT_KEY + "\":" + "\"" + eventValue + "\"}";
+        String idValue = ModuleVersionPublishedEvent.class.equals(event.getClass()) ? this.getPantheonHost() + PANTHEON_MODULE_API_PATH
+                + moduleUUID : "";
+        String uriValue = "";
+        String msg = "";
+        if (System.getenv(PORTAL_URL) != null) {
+            uriValue = System.getenv(PORTAL_URL) + "/topics/" + ServletUtils.toLanguageTag(DEFAULT_MODULE_LOCALE) + "/" + moduleUUID;
+        }
+        if (ModuleVersionPublishedEvent.class.equals(event.getClass())) {
+            msg = "{\""
+                    + ID_KEY + "\":" + "\"" + idValue +"\","
+                    + "\"" + EVENT_KEY + "\":" + "\"" + eventValue + "\"}";
+        } else {
```

Comment:
Ok, I think I understand now.... in this case I would turn this into an `else if` for the specific implementation class, and have the `else` statement log some sort of warning. It will help us identify when we have an event type that is not being handled by the code.